File: С# File Server (React+postges)/File-server/File-server/Entity/FileData.cs
using System.ComponentModel.DataAnnotations;
using System.Xml.Linq;

namespace File_server.Entity
{
    public class FileData
    {
        [Key]
        public long Id { get; set; }
        public string? Name { get; set; }
        public string? RealName{ get; set; }
        public string? Path { get; set; }
        public DateTime? UploadDate { get; set; }
        public string? Type { get; set; }
        public long? Size { get; set; } // в байтах, перевод Кб, Мб будет на фронтенде
        public bool? IsDeleted { get; set; }
    }

    public class FileDto
    {
        public string? Name { get; set; }
        public string? RealName { get; set; }
        public string? Path { get; set; }
        public DateTime? UploadDate { get; set; }
        public string? Type { get; set; }
        public long? Size { get; set; } // в байтах, перевод Кб, Мб будет на фронтенде
        public bool? IsDeleted { get; set; }
    }
}

File: Java File Server - Yandex Disk/src/main/java/gk17/java_file_server__yandex_disk/entity/FileData.java
package gk17.java_file_server__yandex_disk.entity;

import jakarta.persistence.*;
import lombok.Data;

import java.sql.Timestamp;

@Entity
@Table(name = "filedata", schema = "public", catalog = "filesdb")
@Data
public class FileData {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id",nullable = false)
    private Long id;
    @Column(name = "name")
    private String name;

    @Column(name = "path")
    private String path;

    @Column(name = "upload_date")
    private Timestamp uploadDate;

    @Column(name = "type")
    private String type;

    @Column(name = "size")
    private Long size;

    @Column(name = "service_name")
    private String serviceName;

    @Column(name = "parent_path")
    private String parentPath; // путь родительской папки

    @Column(name = "is_deleted")
    private Boolean isDeleted = false; // мягкое удаление
}
